File: spring_java/src/main/java/dev/pollito/spring_java/sakila/film/domain/model/FilmLanguage.java
package dev.pollito.spring_java.sakila.film.domain.model;

import dev.pollito.spring_java.config.enums.ValuedEnum;
import lombok.Getter;
import lombok.RequiredArgsConstructor;

@Getter
@RequiredArgsConstructor
public enum FilmLanguage implements ValuedEnum<String> {
  ENGLISH("English"),
  ITALIAN("Italian"),
  JAPANESE("Japanese"),
  MANDARIN("Mandarin"),
  FRENCH("French"),
  GERMAN("German");

  private final String value;
}

File: spring_kotlin/src/main/kotlin/dev/pollito/spring_kotlin/sakila/film/domain/model/FilmLanguage.kt
package dev.pollito.spring_kotlin.sakila.film.domain.model

import dev.pollito.spring_kotlin.config.enums.ValuedEnum

enum class FilmLanguage(private val value: String) : ValuedEnum<String> {
  ENGLISH("English"),
  ITALIAN("Italian"),
  JAPANESE("Japanese"),
  MANDARIN("Mandarin"),
  FRENCH("French"),
  GERMAN("German");

  override fun getValue(): String = value
}
